hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieTableServiceClient.java

```diff
@@ -734,7 +734,7 @@ protected List<String> getInstantsToRollback(HoodieTableMetaClient metaClient, H
   @Deprecated
   public boolean rollback(final String commitInstantTime, Option<HoodiePendingRollbackInfo> pendingRollbackInfo, boolean skipLocking) throws HoodieRollbackException {
     LOG.info("Begin rollback of instant " + commitInstantTime);
-    final String rollbackInstantTime = pendingRollbackInfo.map(entry -> entry.getRollbackInstant().getTimestamp()).orElse(HoodieActiveTimeline.createNewInstantTime());
+    final String rollbackInstantTime = pendingRollbackInfo.map(entry -> entry.getRollbackInstant().getTimestamp()).orElse(commitInstantTime);
     final Timer.Context timerContext = this.metrics.getRollbackCtx();
```

Comment:
I'm confused, why rollback the instant itself under the current transaction.

Reply:
Hi @danny0405, i just realised that this way of picking timestamp from instant itself will make features like `rollbackToInstant` fail. This approach fails in placing rollback instant in chronological order in timeline. 
I'll resubmit the PR with below approach, 
use approach similar to below method, which actually takes care of rollbacks first before creating instant for the commit.
`org.apache.hudi.client.BaseHoodieWriteClient#startCommit(java.lang.String, org.apache.hudi.common.table.HoodieTableMetaClient)`